### PR TITLE
ISPN-6513 Inconsistencies in Maven pom files

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.jboss.org/browse/ISPN-6513

* Removed repository definitions from pom files as this is generally not recommended
* Added missing modules to dependency management section in the bom file

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -166,6 +181,56 @@
          </dependency>
          <dependency>
             <groupId>${project.groupId}</groupId>
+            <artifactId>infinispan-distexec-demo</artifactId>
+            <version>${project.version}</version>
+         </dependency>
+         <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>infinispan-gridfs-webdav</artifactId>
+            <version>${project.version}</version>
```
Comment (tristantarrant) [inline bom/pom.xml]: We should not include demos and examples in the bom.

Comment (mgencur) [inline bom/pom.xml]: Why? When someone wants to try the demo they probably want the right version of dependencies...

Comment (tristantarrant) [inline bom/pom.xml]: If they want to build the demo, they build it as part of the tree, not import it as a dependency.

Comment (slaskawi) [inline bom/pom.xml]: Actually Martin has a point here. With our [new SNAPSHOT deployment](https://repository.jboss.org/nexus/content/repositories/snapshots/org/infinispan/infinispan-bom/9.0.0-SNAPSHOT) (which deploys new SNAPSHOTs whenever we have a stable build) it is perfectly legal to build demos without other modules (of course whether it's sensible or not is another story...)

**Comment by mgencur** [comment]
Updated.

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**